packages/std/src/mock.rs

```diff
@@ -158,9 +158,21 @@ impl MockQuerier {
 }
 
 impl Querier for MockQuerier {
-    fn query(&self, request: &QueryRequest) -> QuerierResult {
-        match request {
+    fn raw_query(&self, bin_request: &[u8]) -> QuerierResult {
+        // MockQuerier doesn't support Custom, so we ignore it completely here
+        let request: QueryRequest<NoMsg> = match from_slice(bin_request) {
```

Comment:
👍 This nicely shows that there is an unnecessary _typed request -> binary -> typed request_ flow in the mock case, which is needed for the real case.